Automattic/wp-calypso
# SPT: Isolate TemplateSelectorControl component

#### Changes proposed in this Pull Request

Isolate `TemplateSelectorControl` component, and rename to `TemplateSelector`.

While working on #38403, I discovered that `TemplateSelectorControl`'s styling is a bit weird, and doesn't lend itself easily to removing the white bar at the bottom of each item that contains the template name. 

This is due to some `padding-top: 100%`, absolute positioning, translating, etc. I think that we can achieve the same styling in a more semantic way, that will also allow us to conditionally remove the label.

To that end, I'd like to isolate the `TemplateSelectorControl` component and styling, so it becomes easier to modify them, since currently, there's just one big style file in FSE.

This PR renames `TemplateSelectorControl` to `TemplateSelector`, and moves it (and its dependencies, and styling) to a separate folder.

#### Testing instructions

- Verify that Gutenboarding still looks the same as before.
- Build FSE and test it in your WP.com sandbox, and verify that it also still looks and feels the same.
- Run tests in `apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector/test/`:
  - `cd apps/full-site-editing/ && npm ci && npm run test:unit`
